The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7824903046/job/21348220982) [(plain)](https://github.com/rust-lang/rust/commit/f0c6f5a7feb0a2d14a68133c050ec5d56902f961/checks/21348220982/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v4' (SHA:b4ffde65f46336ab88eb53be808477a3936bae11)
Download action repository 'actions/upload-artifact@v3' (SHA:a8a3f3ad30e3422c9c7b888a15615d19a852ae32)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=Alfriadox
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_05a7ca1f-76a1-4cea-8149-376f341c1c51
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=bb674188335f4d3330cc2cdfd9bf9805ec97a079
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_05a7ca1f-76a1-4cea-8149-376f341c1c51
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_05a7ca1f-76a1-4cea-8149-376f341c1c51
GITHUB_TRIGGERING_ACTOR=Alfriadox
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/120764/merge
GITHUB_WORKFLOW_SHA=bb674188335f4d3330cc2cdfd9bf9805ec97a079
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#10 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#10 DONE 0.0s

#11 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#11 0.566   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#11 0.585 Collecting boolean-py==4.0
#11 0.592   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#11 0.609 Collecting chardet==5.1.0
---
#11 3.740 Building wheels for collected packages: reuse
#11 3.741   Building wheel for reuse (pyproject.toml): started
#11 4.064   Building wheel for reuse (pyproject.toml): finished with status 'done'
#11 4.064   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#11 4.065   Stored in directory: /tmp/pip-ephem-wheel-cache-__j3t4yn/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#11 4.067 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#11 4.090   Attempting uninstall: setuptools
#11 4.091     Found existing installation: setuptools 59.6.0
#11 4.092     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#11 4.092     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#11 4.092     Can't uninstall 'setuptools'. No files were found to uninstall.
#11 4.762 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#11 4.762 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#11 5.266 Collecting virtualenv
#11 5.312   Downloading virtualenv-20.25.0-py3-none-any.whl (3.8 MB)
#11 5.478      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 22.9 MB/s eta 0:00:00
#11 5.540 Collecting platformdirs<5,>=3.9.1
#11 5.547   Downloading platformdirs-4.2.0-py3-none-any.whl (17 kB)
#11 5.580 Collecting filelock<4,>=3.12.2
#11 5.587   Downloading filelock-3.13.1-py3-none-any.whl (11 kB)
#11 5.608 Collecting distlib<1,>=0.3.7
#11 5.615   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#11 5.625      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 51.9 MB/s eta 0:00:00
#11 5.710 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#11 5.874 Successfully installed distlib-0.3.8 filelock-3.13.1 platformdirs-4.2.0 virtualenv-20.25.0
#11 DONE 5.9s

#12 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#12 DONE 0.0s
---
DirectMap4k:      176064 kB
DirectMap2M:     8212480 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished dev [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/6894f435d35d3d540dcefbc51390158ca5954861/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-6894f435d35d3d540dcefbc51390158ca5954861-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished release [optimized] target(s) in 25.29s
##[endgroup]
fmt check
##[error]Diff in /checkout/library/core/src/str/mod.rs at line 1161:
     /// function or closure that determines if a character matches.
     ///
     /// Note that there is a footgun to this method when using a slice of [`char`]s.
-    /// Some users may expect that a slice of chars will behave similarly to a `&str` with this method. 
+    /// Some users may expect that a slice of chars will behave similarly to a `&str` with this method.
     /// That is not currently the case. When you pass a slice of [`char`]s to this method, it will return true
     /// if any of the [`char`]s in the slice is the first [`char`] of this string slice. It does not work for
     /// sequentially comparing a slice of [`char`]s to a string slice. See the second example below.
##[error]Diff in /checkout/library/core/src/str/mod.rs at line 1177:
     /// assert!(bananas.starts_with("bana"));
     /// assert!(!bananas.starts_with("nana"));
-    /// 
+    ///
     /// ```
     /// ```
     /// let bananas = "bananas";
-    /// // Note that both of these assert successfully. 
+    ///
+    /// // Note that both of these assert successfully.
+    /// // Note that both of these assert successfully.
     /// assert!(bananas.starts_with(&['b', 'a', 'n', 'a']));
     /// assert!(bananas.starts_with(&['a', 'b', 'c', 'd']));
     /// ```
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/iter/sources/once_with.rs" "/checkout/library/core/src/ascii.rs" "/checkout/library/core/src/primitive.rs" "/checkout/library/core/src/str/mod.rs" "/checkout/library/core/src/str/count.rs" "/checkout/library/core/src/str/traits.rs" "/checkout/library/core/src/str/iter.rs" "/checkout/library/core/src/iter/sources/repeat_n.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Thu Feb  8 04:35:03 UTC 2024
  network time: Thu, 08 Feb 2024 04:35:03 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/str/mod.rs
+++ b/library/core/src/str/mod.rs
@@ -1157,14 +1157,13 @@ impl str {
     ///
     /// Returns `false` if it does not.
     ///
-    /// The [pattern] can be a `&str`, [`char`], a slice of [`char`]s, or a
-    /// function or closure that determines if a character matches.
+    /// The [pattern] can be a `&str`, in which case this function will return true if
+    /// the `&str` is a prefix of this string slice.
     ///
-    /// Note that there is a footgun to this method when using a slice of [`char`]s.
-    /// Some users may expect that a slice of chars will behave similarly to a `&str` with this method.
-    /// That is not currently the case. When you pass a slice of [`char`]s to this method, it will return true
-    /// if any of the [`char`]s in the slice is the first [`char`] of this string slice. It does not work for
-    /// sequentially comparing a slice of [`char`]s to a string slice. See the second example below.
+    /// The [pattern] can also be a [`char`], a slice of [`char`]s, or a
+    /// function or closure that determines if a character matches.
+    /// These will only be checked against the first character of this string slice.
+    /// Look at the second example below regarding behavior for slices of [`char`]s.
     ///
     /// [`char`]: prim@char
     /// [pattern]: self::pattern

--- a/library/core/src/str/mod.rs
+++ b/library/core/src/str/mod.rs
@@ -1161,7 +1161,7 @@ impl str {
     /// function or closure that determines if a character matches.
     ///
     /// Note that there is a footgun to this method when using a slice of [`char`]s.
-    /// Some users may expect that a slice of chars will behave similarly to a `&str` with this method. 
+    /// Some users may expect that a slice of chars will behave similarly to a `&str` with this method.
     /// That is not currently the case. When you pass a slice of [`char`]s to this method, it will return true
     /// if any of the [`char`]s in the slice is the first [`char`] of this string slice. It does not work for
     /// sequentially comparing a slice of [`char`]s to a string slice. See the second example below.
@@ -1177,11 +1177,11 @@ impl str {
     /// assert!(bananas.starts_with("bana"));
     /// assert!(!bananas.starts_with("nana"));
     /// ```
-    /// 
+    ///
     /// ```
     /// let bananas = "bananas";
-    /// 
-    /// // Note that both of these assert successfully. 
+    ///
+    /// // Note that both of these assert successfully.
     /// assert!(bananas.starts_with(&['b', 'a', 'n', 'a']));
     /// assert!(bananas.starts_with(&['a', 'b', 'c', 'd']));
     /// ```

